carousel/core/calculations.py

```diff
@@ -176,31 +83,41 @@ class Calc(object):
     __metaclass__ = CalcBase
 
     def __init__(self):
-        # XXX: Hack to get PR#68 done
-        if not self.parameters:
-            self.parameters = {
-                'always_calc': getattr(self, 'always_calc', False),
-                'frequency': getattr(self, 'frequency', [1, '']),
-                'dependencies': getattr(self, 'dependencies', []),
-                'static': getattr(self, 'static', []),
-                'dynamic': getattr(self, 'dynamic', [])
-            }
+        meta = getattr(self, CalcBase._meta_attr)
+        parameters = getattr(self, CalcBase._param_attr)
         #: ``True`` if always calculated (day and night)
-        self.always_calc = self.parameters.get('always_calc', False)
-        freq = self.parameters.get('frequency', [1, ''])
+        self.always_calc = dict.fromkeys(
```

Comment:
Thanks @anomam check out [L111-L120 in `carousel.core.calculations.Calc`](../blob/953a7d8725e30f39e8ed3ed002a1c657eac4d6f9/carousel/core/calculations.py#L111-L120), it loops over all of the attributes, in the parameters, and overrides, them separately if they exist. So you can either set an attribute for _all_ parameters using `Meta` and/or set individual parameters as needed for convenience.